### PR TITLE
Introduce CompositeBlock

### PR DESCRIPTION
**Pull request by dnhatn**
This change introduces the CompositeBlock to support the aggregated_metrics_double type in ESQL.

### CODE REVIEW

File: x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/ElementType.java [hunk 1]
```diff
@@ -29,6 +29,11 @@ public enum ElementType {
      */
     DOC(DocBlock::newBlockBuilder),
 
+    /**
+     * Composite blocks which contain array of sub-blocks.
+     */
+    COMPOSITE((blockFactory, estimatedSize) -> { throw new UnsupportedOperationException("can't build composite blocks"); }),
```
Comment (nik9000) [inline x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/ElementType.java:35]: :+1:

I could see us wanting some kind of composite element type one day, but we can cross that bridge when we actually need it.

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/physical/EstimatesRowSize.java [hunk 2]
```diff
@@ -117,6 +117,8 @@ static int estimateSize(DataType dataType) {
             case INT -> Integer.BYTES;
             case LONG -> Long.BYTES;
             case NULL -> 0;
+            // TODO: provide a specific estimate for aggregated_metrics_double
+            case COMPOSITE -> throw new EsqlIllegalArgumentException("can't estimate size for composite blocks");
```
Comment (nik9000) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/physical/EstimatesRowSize.java:121]: Fun.

**Review by nik9000 — COMMENTED** [review]
Makes sense to me. Seems like we could get it in safely and build on top.

**Comment by dnhatn** [comment]
@nik9000 Great, let's proceed with the composite block. Thank you for review and feedback.

**Pull request merged**